plugins/catalog/src/components/CatalogPage/utils/timeUtil.test.js

```diff
@@ -14,19 +14,29 @@
  * limitations under the License.
  */
 
-import { isValidDateAndFormat, getTimeBasedGreeting } from './timeUtil';
+import {
+  isValidDate,
+  isValidDateAndFormat,
+  getTimeBasedGreeting,
+} from './timeUtil';
+import { DateTime } from 'luxon';
+
+it('validate isValidDate', () => {
+  expect(isValidDate(DateTime.local().toISO())).toBeTruthy();
+  expect(isValidDate('2021-02-04 15:00:10')).toBeFalsy();
+});
 
 it('validates time format', () => {
   const valid = isValidDateAndFormat(
-    '1970-01-01T00:00:00',
-    'YYYY-MM-DD[T]HH:mm:ss',
+    '1970-01-01 00:00:00',
+    'yyyy-MM-dd hh:mm:ss',
   );
   const invalid = isValidDateAndFormat(
-    '1970/01/01T00:00:00',
-    'YYYY-MM-DD[T]HH:mm:ss',
+    '1970-01-01T00:00:00',
+    'yyyy-MM-ddThh:mm:ss',
   );
-  expect(valid).toBe(true);
-  expect(invalid).toBe(false);
+  expect(valid).toBeTruthy();
+  expect(invalid).toBeFalsy();
```

Comment:
I may b e wrong, but this seems to be changing the point of this test?

Lines 25/26 in the original look to have been set to PURPOSELY set a bad date format which should NOT validate. You seem to have updated the format to match the passed value, and then instead of checking for an actual false value, you're checking only Falsy?  i may not be reading this correctly, but something seems off on these tests.

Reply:
The issue I had when porting these test was that I have not found a way to write a format that accepts a date string that contains `T` since `T` is a token that matches a _localized 24-hour time_ (e.g. 12:00) in Luxon.

The test on line 39 will fail since `yyyy-MM-ddThh:mm:ss` is interpreted by Luxono to be: `yyyy-MM-ddhh:mmhh:mm:ss` which doesn't match `1970-01-01T00:00:00` so we will get an invalid DateTime object and the `isValidDateAndFormat ` will return false.

Reply:
ok!  I'll leave this review to the experts (the core team `backstage/maintainers`) as I'm not an expert in it. :)